test_suite/gap/test_gap_privacy.py

```diff
@@ -73,640 +31,347 @@ def address_is_random_non_resolvable(address):
-    )
+def perform_bonding_with_privacy_and_disconnect(peripheral, central):
+    central_ss, peripheral_ss = init_security_sessions(central, peripheral)
 
+    peripheral.gap.enablePrivacy(True)
+    central.gap.enablePrivacy(True)
+    sleep(0.1)
 
-def perform_bonding(peripheral, central):
-    peripheral.gap.setAdvertisingType("ADV_CONNECTABLE_UNDIRECTED")
-    peripheral.gap.startAdvertising()
+    central.gap.setCentralPrivacyConfiguration(False, "DO_NOT_RESOLVE")
+    central.gap.setPeripheralPrivacyConfiguration(False, "DO_NOT_RESOLVE")
+    peripheral.gap.setPeripheralPrivacyConfiguration(False, "DO_NOT_RESOLVE")
+    peripheral.gap.setCentralPrivacyConfiguration(False, "DO_NOT_RESOLVE")
 
-    peripheral_address_type, peripheral_address = get_peripheral_address(central, peripheral)
-
-    # connect to the device
-    connection_params = get_connection_args(
-        peripheral_address_type,
-        peripheral_address
-    )
-    peripheral_connection = peripheral.gap.waitForConnection.setAsync()(1000)
-    central_connection = central.gap.connect(*connection_params).result
-    peripheral_connection = peripheral_connection.result
-
-    # wait for security manager
-    peripheral_bond = peripheral.securityManager.waitForEvent.setAsync()(
-        peripheral_connection["handle"],
-        15000
-    )
-
-    central_bond = central.securityManager.waitForEvent(
-        central_connection["handle"],
-        15000
-    ).result
-    peripheral_bond = peripheral_bond.result
-
-    central.gap.disconnect(central_connection["handle"], "REMOTE_USER_TERMINATED_CONNECTION")
-
-    # FIXME: needs event from the peripheral and central that they have been
-    # disconnected
-    sleep(1)
+    central_ss.connect(peripheral_ss)
 
+    # Always start peripheral_ss first
+    peripheral_ss.wait_for_event()
+    central_ss.start_pairing()
 
-@pytest.fixture(scope="function")
-def rand_data():
-    rand_data = get_rand_data("MANUFACTURER_SPECIFIC_DATA")
-    # make sure that the data have an acceptable size
-    if len(rand_data) > 16:
-        rand_data = rand_data[0:16]
-    return rand_data
+    # We should get a request
+    peripheral_ss.expect_pairing_request()
 
+    # Accept request, pairing should complete successfully
+    peripheral_ss.accept_pairing_request()
+    peripheral_ss.expect_pairing_success()
 
-@pytest.fixture(scope="function")
-def peripheral(board_allocator: BoardAllocator, rand_data):
-    device = board_allocator.allocate('peripheral')
-    assert device is not None
-    init_device(device)
+    # central_ss should see pairing complete successfully too
+    central_ss.expect_pairing_success()
 
-    configure_peripheral(device, rand_data)
-    yield device
+    central.gap.disconnect(peripheral_ss.connection_handle, "USER_TERMINATION")
 
-    device.ble.shutdown()
-    board_allocator.release(device)
+    # FIXME: needs event from the peripheral and central that they have been disconnected
```

Comment:
Events are present in a queue in the `events` field. However no JSON deserialization has been applied. We should improve this.

Reply:
do you want me to do this as part of this jira task?

Reply:
Jira task is enough. I got a one line _fix_ for ble-cliapp to not output event in the middle of a command response. The task would be about adding JSON deserialization of events.